resources/fetching/fetchers/azure/insights_batch_fetcher_test.go

```diff
@@ -0,0 +1,173 @@
+						Assets:  []inventory.AzureAsset{},
+					},
+				},
+			},
+			expectedErr: false,
+		},
+
+		"2 subs": {
+			mockSubscriptions: map[string]string{"sub1": "subName1", "sub2": "subName2"},
+			mockInventoryAssets: []inventory.AzureAsset{
+				{Id: "id1", Name: "name1", SubscriptionId: "sub1"},
+				{Id: "id2", Name: "name2", SubscriptionId: "sub1"},
+				{Id: "id3", Name: "name3", SubscriptionId: "sub2"},
+			},
+			mockInventoryError: nil,
+			cycleMetadata:      cycle(124),
+			expected: []fetching.ResourceInfo{
+				{
+					CycleMetadata: cycle(124),
+					Resource: &AzureBatchResource{
+						Type:    fetching.MonitoringIdentity,
+						SubType: fetching.AzureDiagnosticSettingsType,
+						SubId:   "sub1",
+						SubName: "subName1",
+						Assets: []inventory.AzureAsset{
+							{Id: "id1", Name: "name1", SubscriptionId: "sub1", SubscriptionName: "subName1"},
+							{Id: "id2", Name: "name2", SubscriptionId: "sub1", SubscriptionName: "subName1"},
+						},
+					},
+				},
+				{
+					CycleMetadata: cycle(124),
+					Resource: &AzureBatchResource{
+						Type:    fetching.MonitoringIdentity,
+						SubType: fetching.AzureDiagnosticSettingsType,
+						SubId:   "sub2",
+						SubName: "subName2",
+						Assets: []inventory.AzureAsset{
+							{Id: "id3", Name: "name3", SubscriptionId: "sub2", SubscriptionName: "subName2"},
+						},
+					},
+				},
+			},
+			expectedErr: false,
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			// fill mock data
+			for i := range tc.mockInventoryAssets {
+				tc.mockInventoryAssets[i].SubscriptionName = tc.mockSubscriptions[tc.mockInventoryAssets[i].SubscriptionId]
+			}
+
+			m := inventory.NewMockServiceAPI(t)
+
+			m.EXPECT().GetSubscriptions().Return(tc.mockSubscriptions).Once()
+			m.EXPECT().ListDiagnosticSettingsAssetTypes(mock.Anything).Return(tc.mockInventoryAssets, tc.mockInventoryError).Once()
+
+			ch := make(chan fetching.ResourceInfo, 100)
+			defer close(ch)
+			f := NewAzureInsightsBatchAssetFetcher(log, ch, m)
+
+			err := f.Fetch(context.Background(), tc.cycleMetadata)
+			if tc.expectedErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+
+			got := testhelper.CollectResources(ch)
+
+			// sort to ensure assertion
+			sortResourceInfoSlice(tc.expected)
+			sortResourceInfoSlice(got)
+
+			assert.Equal(t, tc.expected, got)
+		})
+	}
+}
+
+func sortResourceInfoSlice(r []fetching.ResourceInfo) {
+	for idx := range r {
+		abr, _ := (&r[idx]).Resource.(*AzureBatchResource)
+		sort.Slice(abr.Assets, func(i, j int) bool { return abr.Assets[i].Id > abr.Assets[j].Id })
+	}
+
+	sort.Slice(r, func(i, j int) bool {
+		var x, y string
+
+		ai, _ := (&r[i]).Resource.(*AzureBatchResource)
+		if len(ai.Assets) > 0 {
+			x = ai.Assets[0].SubscriptionId
+		}
+
+		aj, _ := (&r[j]).Resource.(*AzureBatchResource)
+		if len(aj.Assets) > 0 {
+			y = aj.Assets[0].SubscriptionId
+		}
+
+		return x > y
+	})
+}
```

Comment:
```suggestion
			assert.ElementsMatch(t, tc.expected, got)
		})
	}
}
```

Reply:
We have 2 levels of slices. One is the resource and the second is the asset of each resource.